internal/internal_command_state_machine.go

```diff
@@ -1171,7 +1171,7 @@ func (h *commandsHelper) recordLocalActivityMarker(activityID string, details ma
 func (h *commandsHelper) recordMutableSideEffectMarker(mutableSideEffectID string, callCountHint int, data *commonpb.Payloads, dc converter.DataConverter) commandStateMachine {
 	// In order to avoid duplicate marker IDs, we must append the counter to the
 	// user-provided ID
-	mutableSideEffectID = fmt.Sprintf("%v_%v", mutableSideEffectID, h.nextCommandEventID)
+	mutableSideEffectID = fmt.Sprintf("%v_%v", mutableSideEffectID, h.getNextID())
```

Comment:
Can you think of any concerns where fixing this breaks replay for those where replay succeeded before? Is there a case where a successfully replaying workflow will now get a different ID for mutable side effect? Or does it feel like this would only get a different ID in cases where replay was probably broken anyways?

(not that I care _that_ much because mutable side effects are so rarely used and were broken on replay until a few months back)

Reply:
If it did get a different ID, replay would be broken. Depending on what exactly the workflow did after the mutable side effect call it may not have been detected. 

This change should not cause any workflows that were replaying to break.